packages/manager/modules/octavia-load-balancer/src/create/service.js

```diff
@@ -25,18 +29,48 @@ export default class OctaviaLoadBalancerCreateService {
         `/cloud/project/${projectId}/region/${regionName}/network/${privateNetwork.id}/subnet`,
       )
       .then(({ data }) => {
-        const subnets = data.reduce((filtered, subnet) => {
-          if (subnet.gatewayIp) {
+        const subnets = data.map((subnet) => ({
+          ...subnet,
+          displayName: subnet.name
+            ? `${subnet.name} - ${subnet.cidr}`
+            : subnet.cidr,
+        }));
+        return subnets;
+      });
+  }
+
+  getFloatingIps(projectId, regionName) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/region/${regionName}/floatingip`)
+      .then(({ data }) => {
+        const availableFloatingIPs = data.reduce((filtered, floatingIp) => {
+          if (!floatingIp.associatedEntity) {
             filtered.push({
-              ...subnet,
-              displayName: subnet.name
-                ? `${subnet.name} - ${subnet.cidr}`
-                : subnet.cidr,
+              ...floatingIp,
+              displayName: floatingIp.ip,
+              type: FLOATING_IP_TYPE.IP,
             });
           }
           return filtered;
         }, []);
-        return subnets;
+
+        availableFloatingIPs.unshift({
+          ip: null,
+          type: FLOATING_IP_TYPE.NO_IP,
+          displayName: this.$translate.instant(
+            'octavia_load_balancer_create_floating_ip_field_no_floating_ip',
+          ),
+        });
+
+        availableFloatingIPs.unshift({
+          ip: null,
+          type: FLOATING_IP_TYPE.CREATE,
+          displayName: this.$translate.instant(
+            'octavia_load_balancer_create_floating_ip_field_new_floating_ip',
+          ),
+        });
```

Comment:
**nitpick:** Keep service for only fetching data from server. Move this logic to controller to make this function reusable.